recipes/sdf/all/test_package/test_package.c

```diff
@@ -0,0 +1,36 @@
+#include "stdio.h"
+
+#define SDF_IMPLEMENTATION
```

Comment:
Why not export SDF_IMPLEMENTATION and STB_IMAGE_IMPLEMENTATION by cpp_info?

Reply:
These definitions must be defined before only one `include` or it will cause duplicated symbol errors. It permit to compile the implementation. Without them, the header just includes the declarations. Their use is documented, the user must therefore be aware of it. (https://github.com/memononen/SDF)

Reply:
I see, and that's why you can use self.cpp_info.defines, to pass to the compiler when building test_package.

Reply:
Defining them in `cpp_info.defines` create errors linking errors if user include `sdf.h` twice or more. These definitions must be defined before include and only in one C (or C++) file.

A lot of header only libraries works like this. For example, stb doesn't define `STB_IMAGE_IMPLEMENTATION` in `cpp_info.defines` (See [here](https://github.com/conan-io/conan-center-index/blob/master/recipes/stb/all/test_package/test_package.cpp#L1)). Same for Tinyexr [who requires to define `TINYEXR_IMPLEMENTATION` in only one cc file](https://github.com/syoyo/tinyexr#usage).

Reply:
that's interesting. don't they use include guards to prevent things defined twice if you include header twice?

Reply:
include guards doesn't works with multiple translation units. They just ensure that a file isn't included twice in a translation unit. The include guard definition is scoped to the translation unit.

With the SDF, the definition must be defined in a single translation unit and not once in each translation unit.

Reply:
Correct.